File: loader/src/main/java/com/walhalla/mtprotolist/utils/QueryFormat.java
package com.walhalla.mtprotolist.utils;

import com.walhalla.mtprotolist.entity.MtprotoProxy;
import com.walhalla.mtprotolist.webproxy.ProxyInfo;
import com.walhalla.ui.DLog;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.net.IDN;
import java.net.InetAddress;
import java.net.URL;
import java.net.UnknownHostException;
import java.util.List;

public class QueryFormat {
    public static JSONArray getArray0(List<MtprotoProxy> serverList) {
        JSONArray jsonArray = new JSONArray();
        for (MtprotoProxy server : serverList) {
            JSONObject jsonObject = new JSONObject();
            try {
                try {
                    InetAddress address = InetAddress.getByName(server.host);
                    String ipAddress = address.getHostAddress();
                    jsonObject.put("query", ipAddress);
                    jsonObject.put("lang", "EN");//Locale.getDefault().getLanguage()
                    jsonArray.put(jsonObject);
                } catch (UnknownHostException e) {
                    e.printStackTrace();
                }

            } catch (JSONException e) {
                DLog.handleException(e);
            }
        }
        return jsonArray;
    }

    public static JSONArray getQueryArray(List<ProxyInfo> serverList) {
        JSONArray jsonArray = new JSONArray();
        for (ProxyInfo server : serverList) {
            JSONObject jsonObject = new JSONObject();
            try {
                try {
                    String urlString = server.proxyUrl;
                    URL url = new URL(urlString);
                    //String host = url.getHost();
                    String host = IDN.toASCII(url.getHost());

                    InetAddress address = InetAddress.getByName(host);
                    String ipAddress = address.getHostAddress();
                    jsonObject.put("query", ipAddress);
                    jsonObject.put("lang", "EN");//Locale.getDefault().getLanguage()
                    jsonArray.put(jsonObject);
                } catch (Exception e) {
                    DLog.handleException(e);
                }
            } catch (Exception e) {
                DLog.handleException(e);
            }
        }
        return jsonArray;
    }
}

File: loader/src/main/java/com/walhalla/mtprotolist/utils/QueryFormat.kt
package com.walhalla.mtprotolist.utils

import com.walhalla.mtprotolist.entity.MtprotoProxy
import com.walhalla.mtprotolist.webproxy.ProxyInfo
import com.walhalla.ui.DLog.handleException
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import java.net.IDN
import java.net.InetAddress
import java.net.URL
import java.net.UnknownHostException

object QueryFormat {
    fun getArray0(serverList: List<MtprotoProxy>): JSONArray {
        val jsonArray = JSONArray()
        for (server in serverList) {
            val jsonObject = JSONObject()
            try {
                try {
                    val address = InetAddress.getByName(server.host)
                    val ipAddress = address.hostAddress
                    jsonObject.put("query", ipAddress)
                    jsonObject.put("lang", "EN") //Locale.getDefault().getLanguage()
                    jsonArray.put(jsonObject)
                } catch (e: UnknownHostException) {
                    e.printStackTrace()
                }
            } catch (e: JSONException) {
                handleException(e)
            }
        }
        return jsonArray
    }

    fun getQueryArray(serverList: List<ProxyInfo>): JSONArray {
        val jsonArray = JSONArray()
        for (server in serverList) {
            val jsonObject = JSONObject()
            try {
                try {
                    val urlString = server.proxyUrl
                    val url = URL(urlString)
                    //String host = url.getHost();
                    val host = IDN.toASCII(url.host)

                    val address = InetAddress.getByName(host)
                    val ipAddress = address.hostAddress
                    jsonObject.put("query", ipAddress)
                    jsonObject.put("lang", "EN") //Locale.getDefault().getLanguage()
                    jsonArray.put(jsonObject)
                } catch (e: Exception) {
                    handleException(e)
                }
            } catch (e: Exception) {
                handleException(e)
            }
        }
        return jsonArray
    }
}
